### DISCUSSION TITLE
WebRTC works with firefox but not with chrome

### DISCUSSION THREAD

**Discussion by idan123456** [aiortc/aiortc#856] [Q&A]
I've created a program that uses webRTC (using the aiortc library) to video stream from CV2 to a browser and I've implemented the signaling by myself using WebSocket. everything works fine and the answer/offers sends as they should and the connection is made successfully, but the stream only works on Firefox for some reason, for some reason it worked only once on chrome although I didn't change anything.

anyone also encountered something like this?

something worth mentioning:

1. the "icecandidate" event doesn't triggers in the server but the connection is still made successfully because the ice candidates are in the answer from the server.
2. the stream in Firefox has lags for some reason but in the one time it worked in chrome it was smother.
3. Firefox sends empty ice candidates when "icecandidate" event is triggered (chrome icecandidate are ok)

This is the offer sdp that is created with chrome: 

> client.js:49 Client offer SDP: v=0 o=- 7465364294295159094 2 IN IP4 127.0.0.1 s=- t=0 0 a=group:BUNDLE 0 a=extmap-allow-mixed a=msid-semantic: WMS m=video 9 UDP/TLS/RTP/SAVPF 96 97 98 99 100 101 35 36 37 38 102 103 104 105 106 107 108 109 127 125 39 40 41 42 43 44 45 46 47 48 112 113 114 115 116 49 c=IN IP4 0.0.0.0 a=rtcp:9 IN IP4 0.0.0.0 a=ice-ufrag:SRbV a=ice-pwd:6E4su56vURNglbd9oIOn2ViB a=ice-options:trickle a=fingerprint:sha-256 C9:63:09:86:01:67:F3:5E:6C:41:BB:A9:BC:C9:11:99:A0:1F:1F:FD:2E:4D:B3:EC:32:31:06:40:19:2C:74:2C a=setup:actpass a=mid:0 a=extmap:1 urn:ietf:params:rtp-hdrext:toffset a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time a=extmap:3 urn:3gpp:video-orientation a=extmap:4 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01 a=extmap:5 http://www.webrtc.org/experiments/rtp-hdrext/playout-delay a=extmap:6 http://www.webrtc.org/experiments/rtp-hdrext/video-content-type a=extmap:7 http://www.webrtc.org/experiments/rtp-hdrext/video-timing a=extmap:8 http://www.webrtc.org/experiments/rtp-hdrext/color-space a=extmap:9 urn:ietf:params:rtp-hdrext:sdes:mid a=extmap:10 urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id a=extmap:11 urn:ietf:params:rtp-hdrext:sdes:repaired-rtp-stream-id a=recvonly a=rtcp-mux a=rtcp-rsize a=rtpmap:96 VP8/90000 a=rtcp-fb:96 goog-remb a=rtcp-fb:96 transport-cc a=rtcp-fb:96 ccm fir a=rtcp-fb:96 nack a=rtcp-fb:96 nack pli a=rtpmap:97 rtx/90000 a=fmtp:97 apt=96 a=rtpmap:98 VP9/90000 a=rtcp-fb:98 goog-remb a=rtcp-fb:98 transport-cc a=rtcp-fb:98 ccm fir a=rtcp-fb:98 nack a=rtcp-fb:98 nack pli a=fmtp:98 profile-id=0 a=rtpmap:99 rtx/90000 a=fmtp:99 apt=98 a=rtpmap:100 VP9/90000 a=rtcp-fb:100 goog-remb a=rtcp-fb:100 transport-cc a=rtcp-fb:100 ccm fir a=rtcp-fb:100 nack a=rtcp-fb:100 nack pli a=fmtp:100 profile-id=2 a=rtpmap:101 rtx/90000 a=fmtp:101 apt=100 a=rtpmap:35 VP9/90000 a=rtcp-fb:35 goog-remb a=rtcp-fb:35 transport-cc a=rtcp-fb:35 ccm fir a=rtcp-fb:35 nack a=rtcp-fb:35 nack pli a=fmtp:35 profile-id=1 a=rtpmap:36 rtx/90000 a=fmtp:36 apt=35 a=rtpmap:37 VP9/90000 a=rtcp-fb:37 goog-remb a=rtcp-fb:37 transport-cc a=rtcp-fb:37 ccm fir a=rtcp-fb:37 nack a=rtcp-fb:37 nack pli a=fmtp:37 profile-id=3 a=rtpmap:38 rtx/90000 a=fmtp:38 apt=37 a=rtpmap:102 H264/90000 a=rtcp-fb:102 goog-remb a=rtcp-fb:102 transport-cc a=rtcp-fb:102 ccm fir a=rtcp-fb:102 nack a=rtcp-fb:102 nack pli a=fmtp:102 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42001f a=rtpmap:103 rtx/90000 a=fmtp:103 apt=102 a=rtpmap:104 H264/90000 a=rtcp-fb:104 goog-remb a=rtcp-fb:104 transport-cc a=rtcp-fb:104 ccm fir a=rtcp-fb:104 nack a=rtcp-fb:104 nack pli a=fmtp:104 level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=42001f a=rtpmap:105 rtx/90000 a=fmtp:105 apt=104 a=rtpmap:106 H264/90000 a=rtcp-fb:106 goog-remb a=rtcp-fb:106 transport-cc a=rtcp-fb:106 ccm fir a=rtcp-fb:106 nack a=rtcp-fb:106 nack pli a=fmtp:106 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42e01f a=rtpmap:107 rtx/90000 a=fmtp:107 apt=106 a=rtpmap:108 H264/90000 a=rtcp-fb:108 goog-remb a=rtcp-fb:108 transport-cc a=rtcp-fb:108 ccm fir a=rtcp-fb:108 nack a=rtcp-fb:108 nack pli a=fmtp:108 level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=42e01f a=rtpmap:109 rtx/90000 a=fmtp:109 apt=108 a=rtpmap:127 H264/90000 a=rtcp-fb:127 goog-remb a=rtcp-fb:127 transport-cc a=rtcp-fb:127 ccm fir a=rtcp-fb:127 nack a=rtcp-fb:127 nack pli a=fmtp:127 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=4d001f a=rtpmap:125 rtx/90000 a=fmtp:125 apt=127 a=rtpmap:39 H264/90000 a=rtcp-fb:39 goog-remb a=rtcp-fb:39 transport-cc a=rtcp-fb:39 ccm fir a=rtcp-fb:39 nack a=rtcp-fb:39 nack pli a=fmtp:39 level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=4d001f a=rtpmap:40 rtx/90000 a=fmtp:40 apt=39 a=rtpmap:41 H264/90000 a=rtcp-fb:41 goog-remb a=rtcp-fb:41 transport-cc a=rtcp-fb:41 ccm fir a=rtcp-fb:41 nack a=rtcp-fb:41 nack pli a=fmtp:41 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=f4001f a=rtpmap:42 rtx/90000 a=fmtp:42 apt=41 a=rtpmap:43 H264/90000 a=rtcp-fb:43 goog-remb a=rtcp-fb:43 transport-cc a=rtcp-fb:43 ccm fir a=rtcp-fb:43 nack a=rtcp-fb:43 nack pli a=fmtp:43 level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=f4001f a=rtpmap:44 rtx/90000 a=fmtp:44 apt=43 a=rtpmap:45 AV1/90000 a=rtcp-fb:45 goog-remb a=rtcp-fb:45 transport-cc a=rtcp-fb:45 ccm fir a=rtcp-fb:45 nack a=rtcp-fb:45 nack pli a=rtpmap:46 rtx/90000 a=fmtp:46 apt=45 a=rtpmap:47 AV1/90000 a=rtcp-fb:47 goog-remb a=rtcp-fb:47 transport-cc a=rtcp-fb:47 ccm fir a=rtcp-fb:47 nack a=rtcp-fb:47 nack pli a=fmtp:47 profile=1 a=rtpmap:48 rtx/90000 a=fmtp:48 apt=47 a=rtpmap:112 H264/90000 a=rtcp-fb:112 goog-remb a=rtcp-fb:112 transport-cc a=rtcp-fb:112 ccm fir a=rtcp-fb:112 nack a=rtcp-fb:112 nack pli a=fmtp:112 level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=64001f a=rtpmap:113 rtx/90000 a=fmtp:113 apt=112 a=rtpmap:114 red/90000 a=rtpmap:115 rtx/90000 a=fmtp:115 apt=114 a=rtpmap:116 ulpfec/90000 a=rtpmap:49 flexfec-03/90000 a=rtcp-fb:49 goog-remb a=rtcp-fb:49 transport-cc a=fmtp:49 repair-window=10000000

This is the offer sdp that is created with Firefox:

> 
> Client offer SDP: v=0
> 
> o=mozilla...THIS_IS_SDPARTA-99.0 2908353415788641642 0 IN IP4 0.0.0.0
> 
> s=-
> 
> t=0 0
> 
> a=fingerprint:sha-256 32:01:6A:3C:45:DF:F3:09:25:E2:F0:97:09:18:3B:9B:30:9A:F8:C8:B3:99:FA:17:F5:32:1D:23:3E:C0:14:FE
> 
> a=group:BUNDLE 0
> 
> a=ice-options:trickle
> 
> a=msid-semantic:WMS *
> 
> m=video 9 UDP/TLS/RTP/SAVPF 120 124 121 125 126 127 97 98
> 
> c=IN IP4 0.0.0.0
> 
> a=recvonly
> 
> a=extmap:3 urn:ietf:params:rtp-hdrext:sdes:mid
> 
> a=extmap:4 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
> 
> a=extmap:5 urn:ietf:params:rtp-hdrext:toffset
> 
> a=extmap:6/recvonly http://www.webrtc.org/experiments/rtp-hdrext/playout-delay
> 
> a=extmap:7 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
> 
> a=fmtp:126 profile-level-id=42e01f;level-asymmetry-allowed=1;packetization-mode=1
> 
> a=fmtp:97 profile-level-id=42e01f;level-asymmetry-allowed=1
> 
> a=fmtp:120 max-fs=12288;max-fr=60
> 
> a=fmtp:124 apt=120
> 
> a=fmtp:121 max-fs=12288;max-fr=60
> 
> a=fmtp:125 apt=121
> 
> a=fmtp:127 apt=126
> 
> a=fmtp:98 apt=97
> 
> a=ice-pwd:e3e05c052f9129605eaa961e614d0a0f
> 
> a=ice-ufrag:4bcf9f69
> 
> a=mid:0
> 
> a=rtcp-fb:120 nack
> 
> a=rtcp-fb:120 nack pli
> 
> a=rtcp-fb:120 ccm fir
> 
> a=rtcp-fb:120 goog-remb
> 
> a=rtcp-fb:120 transport-cc
> 
> a=rtcp-fb:121 nack
> 
> a=rtcp-fb:121 nack pli
> 
> a=rtcp-fb:121 ccm fir
> 
> a=rtcp-fb:121 goog-remb
> 
> a=rtcp-fb:121 transport-cc
> 
> a=rtcp-fb:126 nack
> 
> a=rtcp-fb:126 nack pli
> 
> a=rtcp-fb:126 ccm fir
> 
> a=rtcp-fb:126 goog-remb
> 
> a=rtcp-fb:126 transport-cc
> 
> a=rtcp-fb:97 nack
> 
> a=rtcp-fb:97 nack pli
> 
> a=rtcp-fb:97 ccm fir
> 
> a=rtcp-fb:97 goog-remb
> 
> a=rtcp-fb:97 transport-cc
> 
> a=rtcp-mux
> 
> a=rtcp-rsize
> 
> a=rtpmap:120 VP8/90000
> 
> a=rtpmap:124 rtx/90000
> 
> a=rtpmap:121 VP9/90000
> 
> a=rtpmap:125 rtx/90000
> 
> a=rtpmap:126 H264/90000
> 
> a=rtpmap:127 rtx/90000
> 
> a=rtpmap:97 H264/90000
> 
> a=rtpmap:98 rtx/90000
> 
> a=setup:actpass
> 
> a=ssrc:3025287394 cname:{6774285f-635b-403e-9325-8ca122f72269}
> 
> client.js:49:11

**Comment by cats256** [comment]
Currently experiencing this. Did you ever figure it out? @idan123456
